samples/subsys/display/lvgl/src/main.c

```diff
@@ -125,12 +149,38 @@ int main(void)
 
 	if (IS_ENABLED(CONFIG_LV_Z_POINTER_KSCAN) || IS_ENABLED(CONFIG_LV_Z_POINTER_INPUT)) {
 		lv_obj_t *hello_world_button;
+		lv_obj_t *canvas = NULL;
+
+#ifdef CONFIG_SAMPLE_LVGL_TOUCH_DRAWING
+		/**
```

Comment:
Would it make the `main` method maybe a bit more readable to extract this into a separate function?